catalog/rest/catalog-rest-endpoint/src/main/java/org/codice/ddf/endpoints/rest/RESTEndpoint.java

```diff
@@ -840,10 +885,18 @@ public Response addDocument(
                           createInfo.getMetacard())),
                   null);
           createResponse = catalogFramework.create(streamCreateRequest);
+          if (createResponse == null
+              || CollectionUtils.isEmpty(createResponse.getCreatedMetacards())) {
+            String errorMessage = "Cannot create at least one catalog entry.";
+            LOGGER.info(errorMessage);
+            return createBadRequestResponse(errorMessage);
+          }
+          // TODO the convention I've been following is that the response header contains the
```

Comment:
This was meant to be a design question for @pklinef, but he has yet to respond.

Reply:
We need to send all created IDs back to the client.  We have never sent more than a single ID in the response header.  It might be safer to have another header for child IDs.  But we will need a way to pass back which metacard is the parent and should be used for the existing response header.